### PR TITLE
WRN-15314 - Migrate Global knobs for accent/highlight/show default skins to controls

### PR DESCRIPTION
**Pull request by vladut-clapou-lgp**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [X] Documentation was verified or is not changed
* [X] UI test was passed or is not needed
* [X] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Implemented custom Storybook addon for Accent, Highlight and Show default skins controls

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-15314

### Comments
Enact-DCO-1.0-Signed-off-by: Clapou George Vladut vladut.clapou@lgepartner.com

### CODE REVIEW

File: samples/sampler/custom-addon/DefaultSkinToolbarTab.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'; // eslint-disable-line
+import {useGlobals} from '@storybook/api';
+import ToolbarTab from './ToolbarTab';
+import PropTypes from 'prop-types';
```
Comment (adrian-cocoara-lgp) [inline samples/sampler/custom-addon/DefaultSkinToolbarTab.js:4]: imports should be grouped as module imports, and below are the local imports.

```suggestion
import {useGlobals} from '@storybook/api';
import PropTypes from 'prop-types';

import ToolbarTab from './ToolbarTab';
```

React import is not necessary. 
Apply this suggestion on the other files as well

File: samples/sampler/custom-addon/ColorPicker.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, {useEffect} from 'react'; // eslint-disable-line
```
Comment (adrian-cocoara-lgp) [inline samples/sampler/custom-addon/ColorPicker.js:1]:
```suggestion
import {useEffect} from 'react';
```

File: samples/sampler/custom-addon/ColorPicker.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import {useGlobals} from '@storybook/api';
+import PropTypes from 'prop-types';
```
Comment (adrian-cocoara-lgp) [inline samples/sampler/custom-addon/ColorPicker.js:3]: add an empty line

Comment (adrian-cocoara-lgp) [reply samples/sampler/custom-addon/ColorPicker.js:3]:
```suggestion
import {useGlobals} from '@storybook/api';
import PropTypes from 'prop-types';
import React, {useEffect} from 'react'; // eslint-disable-line
```

File: samples/sampler/custom-addon/Toolbar.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import React, {memo} from 'react'; // eslint-disable-line
+import {WithTooltip, TooltipLinkList} from '@storybook/components';
+import {useGlobals} from '@storybook/api';
+import ColorPicker from './ColorPicker';
+import ToolbarTab from './ToolbarTab';
+import DefaultSkinToolbarTab from './DefaultSkinToolbarTab';
```
Comment (adrian-cocoara-lgp) [inline samples/sampler/custom-addon/Toolbar.js:6]: apply the suggestion I mentioned above

File: samples/sampler/custom-addon/ToolbarTab.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'; // eslint-disable-line
+import PropTypes from 'prop-types';
+import {Icons, IconButton} from '@storybook/components';
```
Comment (adrian-cocoara-lgp) [inline samples/sampler/custom-addon/ToolbarTab.js:3]:
```suggestion
import {Icons, IconButton} from '@storybook/components';
import PropTypes from 'prop-types';
import React from 'react'; // eslint-disable-line
```

File: samples/sampler/custom-addon/register.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'; // eslint-disable-line
+import {addons, types} from '@storybook/addons';
```
Comment (adrian-cocoara-lgp) [inline samples/sampler/custom-addon/register.js:2]: sort imports

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: samples/sampler/custom-addon/DefaultSkinToolbarTab.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+const DefaultSkinToolbarTab = ({toolbarParamKey}) => {
+	const [globals, updateGlobals] = useGlobals();
+	const isActive = globals[toolbarParamKey] || false;
+
+	const toggleState = () => {
+		updateGlobals({
+			[toolbarParamKey]: !isActive
+		});
+	};
+	return (
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/DefaultSkinToolbarTab.js:16]: Please leave empty line above return().

File: samples/sampler/custom-addon/Toolbar.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import {useGlobals} from '@storybook/api';
+import {WithTooltip, TooltipLinkList} from '@storybook/components';
+import React, {memo} from 'react'; // eslint-disable-line
+
+import ColorPicker from './ColorPicker';
+import DefaultSkinToolbarTab from './DefaultSkinToolbarTab';
+import ToolbarTab from './ToolbarTab';
+
+const getToolTipLink = (isColorPicker, colorPickerType) => {
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/Toolbar.js:9]: Please order params alphabetically. Below as well.

File: samples/sampler/custom-addon/ToolbarTab.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import PropTypes from 'prop-types';
+import React from 'react'; // eslint-disable-line
+
+const ToolbarTab = ({toolbarParamKey, isActive, toggleState}) => {
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/ToolbarTab.js:5]: Please order params alphabetically.

File: samples/sampler/custom-addon/ToolbarTab.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+	return (
+		<IconButton
+			key={toolbarParamKey}
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/ToolbarTab.js:8]: Please order props.

File: samples/sampler/custom-addon/Toolbar.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+	}
+};
+
+const getColorPickerTab = (isColorPicker, defaultSkins, toolbarParamKey) => {
+	const toolTipLink = [getToolTipLink(isColorPicker, toolbarParamKey)];
+
+	if (defaultSkins) return null;
+	return (
+		<WithTooltip
+			placement="top"
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/Toolbar.js:28]: Please order props.

File: samples/sampler/custom-addon/register.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+import React from 'react'; // eslint-disable-line
+
+import {ACCENT_ADDON_ID, ACCENT_PARAM_KEY, DEFAULTSKINS_ADDON_ID, DEFAULTSKINS_PARAM_KEY, HIGHLIGHT_ADDON_ID, HIGHLIGHT_PARAM_KEY} from './constants';
+import Toolbar from './Toolbar';
+
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/register.js:6]: Please remove the extra empty line.

File: samples/sampler/custom-addon/register.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+	const renderDefaultSkinsTab = () => <Toolbar param={DEFAULTSKINS_PARAM_KEY} isColorPicker={false} />;
+
+	addons.add(ACCENT_ADDON_ID, {
+		title: 'Accent tab',
+		type: types.TOOL,
+		match: ({viewMode}) => !!(viewMode && viewMode.match(/^(story|docs)$/)),
+		render: renderAccentColorPickerTab,
+		paramKey: ACCENT_ADDON_ID
+	});
+	addons.add(HIGHLIGHT_ADDON_ID, {
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/register.js:20]: Please add empty line above.

File: samples/sampler/custom-addon/register.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+		title: 'Highlight tab',
+		type: types.TOOL,
+		match: ({viewMode}) => !!(viewMode && viewMode.match(/^(story|docs)$/)),
+		render: renderHighlightColorPickerTab,
+		paramKey: HIGHLIGHT_PARAM_KEY
+	});
+	addons.add(DEFAULTSKINS_ADDON_ID, {
```
Comment (stanca-pop-lgp) [inline samples/sampler/custom-addon/register.js:27]: Please add empty line above.

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Pull request merged**